From bc5afdac354269412903bc482d879c2b405462bf Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Fri, 08 Apr 2022 09:58:59 +0300
Subject: [PATCH] [SW-79695] gaudi3: update efuse bit map

Efuse bitmap has been updated for gaudi3. The patch updates
the bitmaps as per requirement.

Change-Id: I2adabce6c4a78c581ec5ec329d7cb92388e8ef59
---

diff --git a/arch/gaudi2/arc/dev_efuse.h b/arch/gaudi2/arc/dev_efuse.h
index 898eb8e..c92d21a 100644
--- a/arch/gaudi2/arc/dev_efuse.h
+++ b/arch/gaudi2/arc/dev_efuse.h
@@ -23,6 +23,9 @@
 #define EFUSE_SYSCFG_FW_VERSION_LOCATION_SPI		BIT(8)
 #define EFUSE_SYSCFG_REVOKATION_LOCATION_SPI		BIT(8)
 
+#define EFUSE_SYSCFG_BACKUP_DISABLE		BIT(20)
+#define EFUSE_SYSCFG_DISABLE_TWI_IF_MASK	BIT(19)
+
 /*
  * As per recommendation from Security Architect, if FUSE is not available,
  * card should not continue in the secure boot mode. Hence the default
diff --git a/arch/gaudi3/arc/dev_efuse.h b/arch/gaudi3/arc/dev_efuse.h
index b785bf9..51f63ac 100644
--- a/arch/gaudi3/arc/dev_efuse.h
+++ b/arch/gaudi3/arc/dev_efuse.h
@@ -9,16 +9,14 @@
 /* Unused for gaudi3. Maintained for common code compilation */
 #define EFUSE_KEY_HASH_OFFSET				0x0
 
-/*
- * 00: ROM, 01: EROM, 10: RSVD, 11:ROM
- */
-#define EFUSE_SYSCFG_RECOVERY_CODE_LOCATION		13
-#define EFUSE_SYSCFG_RECOVERY_CODE_LOCATION_MASK	0x3
-#define EFUSE_SYSCFG_RECOVERY_CODE_LOCATION_EROM	0x1
-
-#define EFUSE_SYSCFG_ENABLE_TPM_MASK			BIT(11)
-
-#define EFUSE_SYSCFG_EFUSE_LOCATION_MASK		BIT(25)
+#define EFUSE_SYSCFG_EOM_MASK				BIT(5)
+#define EFUSE_SYSCFG_TPM_CODE_LOCATION_MASK		BIT(6)
+#define EFUSE_SYSCFG_RECOVERY_CODE_LOCATION_MASK	BIT(7)
+#define EFUSE_SYSCFG_DISABLE_TWI_IF_MASK		BIT(11)
+#define EFUSE_SYSCFG_EFUSE_LOCATION_MASK		BIT(13)
+#define EFUSE_SYSCFG_SECURITY_ENABLE_MASK		BIT(15)
+#define EFUSE_SYSCFG_ENABLE_TPM_MASK			BIT(17)
+#define EFUSE_SYSCFG_BACKUP_DISABLE			BIT(22)
 #define EFUSE_SYSCFG_SKU_MASK				BIT(26)
 
 /* Following bitmaps are used in common code, hence maintained for compilation */
diff --git a/arch/gaudi3/arc/recovery.c b/arch/gaudi3/arc/recovery.c
index 161422c..3e7daf9 100644
--- a/arch/gaudi3/arc/recovery.c
+++ b/arch/gaudi3/arc/recovery.c
@@ -185,8 +185,6 @@
 
 void execute_recovery(void)
 {
-	uint8_t recovery_location;
-
 	arcboot_rom_status_update(PROGRESS_ARCBOOT_IN_RECOVERY);
 
 	/* Notify slave if master die and dual die sku */
@@ -194,18 +192,11 @@
 		arcboot_rom_notify_slave();
 
 	/*
-	 * efuse[14:13]:
-	 * 00: ROM
-	 * 01: eROM
-	 * 10: Reserved
-	 * 11: ROM
+	 * efuse[7]:
+	 * 0: eROM
+	 * 1: ROM
 	 */
-	recovery_location = ((arcboot_rom_global.config_fuse >>
-			EFUSE_SYSCFG_RECOVERY_CODE_LOCATION) &
-			EFUSE_SYSCFG_RECOVERY_CODE_LOCATION_MASK);
-
-	if (recovery_location != EFUSE_SYSCFG_RECOVERY_CODE_LOCATION_EROM)
-		/* In all other scenarios, execute from ROM (most secure) */
+	if (arcboot_rom_global.config_fuse & EFUSE_SYSCFG_RECOVERY_CODE_LOCATION_MASK)
 		execute_rom_recovery();
 	else
 		execute_erom_recovery();
diff --git a/arch/greco/arc_elf32/dev_efuse.h b/arch/greco/arc_elf32/dev_efuse.h
index bc08814..c2963a0 100644
--- a/arch/greco/arc_elf32/dev_efuse.h
+++ b/arch/greco/arc_elf32/dev_efuse.h
@@ -16,6 +16,9 @@
 #define EFUSE_SYSCFG_KEY_LOCATION_SPI		BIT(8)
 #define EFUSE_SYSCFG_REVOKATION_LOCATION_SPI	BIT(6)
 
+#define EFUSE_SYSCFG_BACKUP_DISABLE		BIT(20)
+#define EFUSE_SYSCFG_DISABLE_TWI_IF_MASK	BIT(19)
+
 #define EFUSE_SYSCFG_DEFAULT	(EFUSE_SYSCFG_SPI_CONTROLLER | \
 				 EFUSE_SYSCFG_ENABLE_VALIDATION_MASK | \
 				 EFUSE_SYSCFG_FW_VERSION_LOCATION_SPI | \
diff --git a/common/efuse.h b/common/efuse.h
index abd3827..47316a3 100644
--- a/common/efuse.h
+++ b/common/efuse.h
@@ -21,8 +21,6 @@
 #define EFUSE_SYSCFG_WA_SNPS_CASE_01038891	BIT(23)
 #define EFUSE_SYSCFG_DISABLE_ROM_PCIE_RECOVERY	BIT(22)
 #define EFUSE_SYSCFG_DISABLE_SECONDARY_IMG	BIT(21)
-#define EFUSE_SYSCFG_BACKUP_DISABLE		BIT(20)
-#define EFUSE_SYSCFG_DISABLE_TWI_IF_MASK	BIT(19)
 #define EFUSE_SYSCFG_PCIE_GEN_LIMIT		BIT(18)
 #define EFUSE_SYSCFG_SPI_CONTROLLER		BIT(17)
 #define EFUSE_SYSCFG_DISABLE_JTAG_IF_MASK	BIT(12)
